### ISSUE TITLE
It's time to consider translating.

### ISSUE THREAD

**Issue by MarisaKirisame** [rustcc/RustPrimer#197]
Now since the book is in a pretty good shape, we should consider whether and how should we translate to English.

The most important question is: why? What's the unique property that make this book stand out from all the other tutorial, thus worth a translation, and become more then a mere localization for Chinese programmer?

The second question is, how? I personally disagree with mike's proposal of waiting for a native tongue, because I suspect that it is pretty rare to find a native English speaker who can speak Chinese, know rust well and care to do the work. OTOH, I believe that basically everyone who contribute more or less know English. IMO it is far more doable to just translate it, and ask native English speaker to do word smithing.

The third problem is, if we translate, what should people use in issue/pull request? English, Chinese, or what they like?

**Comment by tennix** [comment]
I agree that we should do it other than waiting for English speaker. But TBH this book was just finished the first edition, lots of bugs should be fixed and lots of revision work to do for the Chinese version. So I suggest we wait for a couple of weeks until we get a stable version.

For why, as I said in the [reddit post](https://www.reddit.com/r/rust/comments/4du8eb/rustprimer_a_rust_book_for_newbies_only_chinese/) this book is written from a normal programmer's perspective. It's more easy to understand for new rust programmers. Besides since this is written by our whole community, we tried to make it as complete as possible. AFAIK the most comprehensive tutorial or book is the [official book](https://doc.rust-lang.org/stable/book), but it's written from Rust team's perspective.

As for how, we(our Chinese community) should translate it first. And for an international project, we should manage this repo more formally and make use of GitHub functionality. Issues should be labeled for English and Chinese. For English translation, commit messages, PRs and issues should be in English

**Comment by wayslog** [comment]
Sounds good and need help !

**Comment by wayslog** [comment]
I  have make a new branch and milestone for your translation
